### PR TITLE
Change additional output type & skip empty optional arguments

### PR DESCRIPTION
**Pull request by akash-akya**
- Change type of additional output values to map from keyword list
- Skip empty optional argument from function definition when the operation does not have any

Hi @kipcole9! I am making few breaking changes, if possible can you check if this breaks `Image`? 
Btw, I also made few other changes in the master - https://github.com/akash-akya/vix/pull/103. Notably, added limited math operators similar to Image :)

### CODE REVIEW

**Comment by kipcole9** [comment]
@akash-akya, I tried `image` with `vix` master branch and all tests pass without change. So far so good! I will very happily ditch my math operators in favour of yours - but I haven't tested that yet.

**Comment by akash-akya** [comment]
Thanks for quick check! 

> I will very happily ditch my math operators in favour of yours - but I haven't tested that yet.

👍🏼. Makes sense to just wait a bit more for it to be stable. I am thinking if we need something like `Nx.Defn`, but not finding strong reason to add that complexity yet.

**Comment by kipcole9** [comment]
TLDR; Some small incompatibilities I can work around.

I spoke too soon because I tested on master before you merge.  Some small issue I can easily work around.
* `Operation.find_trim/2` now doesn't return `flags` as the last element in the return tuple (easy fix for me, no issue)
* `Vix.Vips.MutableOperation.draw_line/7` is now `Vix.Vips.MutableOperation.draw_line/6` which should be easy to work around
* `Operation.profile/1` now doesn't return flags in the return tuple - also easily worked around in my code.

There is one other failing tests but the cause is not immediately clear. It is coming from  `Image.skew_angle/1` meaning its either `Operation.fft` or `Operation.project` that has a different return type I have to investigate.

**Comment by akash-akya** [comment]
I see. Let me know if you need any help with that, I can create a PR.

> There is one other failing tests but the cause is not immediately clear. It is coming from Image.skew_angle/1 meaning its either Operation.fft or Operation.project that has a different return type I have to investigate.

It should be `project`, same as point one.  Previously we returned empty list as last value, now we don't

**Comment by kipcole9** [comment]
All good Akash, they are easy fixes and a cleaner API so happy to push another `Image` release when you ship yours.

Do you have a view on when you please to release this version of `Vix`?

**Comment by akash-akya** [comment]
Well then nothing is pending from my side, I'll create a release now.

**Comment by kipcole9** [comment]
Sounds good, I'll get my lib back in shape after you publish and do the same.  Glad you're still motivate by working on `vix` - its an amazing library and, if I can say, beautifully done.

**Comment by akash-akya** [comment]
Thanks! Yes, still have a lot of motivation, only wish I had more time to spare, anw I'll complain about things some other day. 

Few things I have on my mind in no order: 

* I want to add support for `VipsRegion` which should make certain tasks very efficient (such as machine learning). Some details about this here: https://github.com/libvips/pyvips/issues/100

* A **_lot_** of improvement needed on documentation/cheat sheet/tutorial/livebooks side. This will be a good exercise for me as well. This is something I can try to learn from your libraries for sure. 

* Add support for blocking image operations: https://www.libvips.org/2022/05/28/What's-new-in-8.13.html

* Explore the possibility of improving discoverability libvips operation, while keeping it maintainable, so that I don't strugle when there is a new libvips version.

I am open to some other important changes as well if you have something on mind.

**Comment by akash-akya** [comment]
@kipcole9 `v0.17.0` is released 🎉

**Comment by kipcole9** [comment]
I'm very interested in where to you get to with those "roadmap" items - very interesting!

`image 0.29.0` is published now with support for `vix 0.17.0`.
